pkg/controllers/user/alert/watcher/event.go

```diff
@@ -0,0 +1,62 @@
+package watcher
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/rancher/rancher/pkg/controllers/user/alert/manager"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type EventWatcher struct {
+	eventLister        v3.ClusterEventLister
+	clusterAlertLister v3.ClusterAlertLister
+	alertManager       *manager.Manager
+	clusterName        string
+}
+
+func StartEventWatcher(cluster *config.UserContext, manager *manager.Manager) {
+	clusterEvents := cluster.Management.Management.ClusterEvents("")
+
+	eventWatcher := &EventWatcher{
+		eventLister:        clusterEvents.Controller().Lister(),
+		clusterAlertLister: cluster.Management.Management.ClusterAlerts(cluster.ClusterName).Controller().Lister(),
+		alertManager:       manager,
+		clusterName:        cluster.ClusterName,
+	}
+
+	clusterEvents.AddClusterScopedHandler("cluster-event-alerter", cluster.ClusterName, eventWatcher.Sync)
```

Comment:
Note that clusterEvents are regular events of k8s cluster backpropagated to management layer so it is easier to list using APIs (https://github.com/rancher/rancher/blob/master/pkg/controllers/user/eventssyncer/eventssyncer.go handles the back propagation). Is there any specific use case requiring listening to cluster events vs listening to regular events instead, in user cluster (workload.Core.Events)?